client/object_put.go

```diff
@@ -264,3 +269,92 @@ func (c *Client) ObjectPutInit(ctx context.Context, prm PrmObjectPutInit) (*Obje
 
 	return &w, nil
 }
+
+type objectWriter struct {
+	context context.Context
+	client  *Client
+}
+
+func (x *objectWriter) InitDataStream(header object.Object) (io.Writer, error) {
+	var prm PrmObjectPutInit
+
+	stream, err := x.client.ObjectPutInit(x.context, prm)
+	if err != nil {
+		return nil, fmt.Errorf("init object stream: %w", err)
+	}
+
+	if stream.WriteHeader(header) {
+		return &payloadWriter{
+			stream: stream,
+		}, nil
+	}
+
+	res, err := stream.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	return nil, apistatus.ErrFromStatus(res.Status())
+}
+
+type payloadWriter struct {
+	stream *ObjectWriter
+}
+
+func (x *payloadWriter) Write(p []byte) (int, error) {
+	if !x.stream.WritePayloadChunk(p) {
+		return 0, x.Close()
+	}
+
+	return len(p), nil
+}
+
+func (x *payloadWriter) Close() error {
+	res, err := x.stream.Close()
+	if err != nil {
+		return err
+	}
+
+	return apistatus.ErrFromStatus(res.Status())
+}
+
+// CreateObjectWithECDSAKey derives user.ID from provided ecdsa.PrivateKey and
+// calls CreateObject.
+func CreateObjectWithECDSAKey(ctx context.Context, cli *Client, key ecdsa.PrivateKey, cnr cid.ID, data io.Reader, attributes ...string) (oid.ID, error) {
+	var owner user.ID
+	user.IDFromKey(&owner, key.PublicKey)
+
+	return CreateObject(ctx, cli, neofsecdsa.Signer(key), cnr, owner, data, attributes...)
+}
+
+// CreateObject creates new NeoFS object with given payload data and stores it
+// in specified container of the NeoFS network using provided Client connection.
+// The object is created on behalf of provided neofscrypto.Signer, and owned by
+// the specified user.ID.
+//
+// In terms of NeoFS, parameterized neofscrypto.Signer represents object owner,
+// object signer and request sender. Container SHOULD be public-write or sender
+// SHOULD have corresponding rights.
+//
+// Client connection MUST be opened in advance, see Dial method for details.
+// Network communication is carried out within a given context, so it MUST NOT
+// be nil.
+func CreateObject(ctx context.Context, cli *Client, signer neofscrypto.Signer, cnr cid.ID, owner user.ID, data io.Reader, attributes ...string) (oid.ID, error) {
```

Comment:
it uses client API, so there is no need to make it a method. Later it can be easily transformed to method

Reply:
Once it's 1.0, it won't be so easy. Even though this doesn't require private `Client` methods it's inherently bound to client's context (requires a proper initialized connection), I think it's just easier to use this way. @smallhive, what do you think?

Reply:
Both variants have a right to live. Separate function maybe even more convenient for testing purposes. Despite it in this case I consider to have the `CreateObject func` as a part of a `Client`.

In current realization all methods to work with objects and containers are connected to client. It is fluent to use them like:
- client.ContainerList
- client.ContainerGet
- client.ObjectHead
- client.ObjectDelete
- etc

Separate function for object creation creates another way to work with SDK and it may confuse our users. I don't mind to change SDK interface, but in future and in another issue. Right now similarity is better, from my point of view

Reply:
Separating these things from `Client` only makes sense if we're separating the raw NeoFS API from higher-level functions (related to #342 and #380).

Reply:
done